File: 0043-multiply-strings/0043-multiply-strings.js
/**
 * @param {string} num1
 * @param {string} num2
 * @return {string}
 */
var multiply = function(num1, num2) {
    if (num1 === '0' || num2 === '0') {
        return '0';
    }
    
    let m = num1.length, n = num2.length;
    let pos = new Array(m + n).fill(0);
   
    for(let i = m - 1; i >= 0; i--) {
        for(let j = n - 1; j >= 0; j--) {
            let mul = (num1[i] - '0') * (num2[j] - '0'); 
            let p1 = i + j, p2 = i + j + 1;
            let sum = mul + pos[p2];

            pos[p1] += Math.floor(sum / 10);
            pos[p2] = sum % 10;
        }
    }  
    
    while (pos[0] === 0) {
        pos.shift();
    }
    
    return pos.join('');
};

File: 0043-multiply-strings/0043-multiply-strings.cs
public class Solution {
    public string Multiply(string num1, string num2) {
        int m = num1.Length, n = num2.Length;
        int[] pos = new int[m + n];
   
        for(int i = m - 1; i >= 0; i--) {
            for(int j = n - 1; j >= 0; j--) {
                int mul = (num1[i] - '0') * (num2[j] - '0'); 
                int p1 = i + j, p2 = i + j + 1;
                int sum = mul + pos[p2];

                pos[p1] += sum / 10;
                pos[p2] = sum % 10;
            }
        }  

        StringBuilder sb = new StringBuilder();
        foreach(int p in pos) if(!(sb.Length == 0 && p == 0)) sb.Append(p);
        return sb.Length == 0 ? "0" : sb.ToString();
    }
}
